pkg/volume/gce_pd/gce_pd.go

```diff
@@ -122,12 +126,50 @@ func (plugin *gcePersistentDiskPlugin) newCleanerInternal(volName string, podUID
 	}}, nil
 }
 
+func (plugin *gcePersistentDiskPlugin) NewDeleter(spec *volume.Spec) (volume.Deleter, error) {
+	return plugin.newDeleterInternal(spec, &GCEDiskUtil{})
+}
+
+func (plugin *gcePersistentDiskPlugin) newDeleterInternal(spec *volume.Spec, manager pdManager) (volume.Deleter, error) {
```

Comment:
Why is the helper function needed? Why not just stick this logic in `NewDeleter`? Ditto with `NewProvisioner`.